src/main/java/org/apache/commons/lang3/EnumUtils.java

```diff
@@ -294,6 +295,25 @@
         return map;
     }
 
+    /**
+     * <p>Gets the {@code Map} of enums.</p>
+     *
+     * <p>This method is useful when you need a map of enums.</p>
+     *
+     * @param <E> the type of enumeration
+     * @param <K> the type of the map key
+     * @param enumClass the class of the enum to query, not null
+     * @param keyFunction the function to query for the key, not null
+     * @return the modifiable map of enums, never null
+     */
```

Comment:
Hi @MaxwellCody and thank you for your PR. Some comments:

- Add `@since 3.13.0`
- "Eat own dog food" and reimplement `EnumUtils.getEnumMap(Class<E>)` to use this method.